lib/puppet/functions/default_content.rb

```diff
@@ -1,10 +1,16 @@
 Puppet::Functions.create_function(:default_content) do
-  # @summary Takes an optional content and an optional template name and
-  # returns the contents of a file.
-  # @param :content
-  # @param :template_name
+  # @summary Takes an optional content and an optional template name and returns the contents of a file.
+  #
+  # @param content
+  # @param template_name
+  #   The path to an .erb or .epp template file or `undef`.
   # @return [String]
-  # @example:
+  #   Returns the value of the content parameter if it's a non empty string.
+  #   Otherwise returns the rendered output from `template_name`.
+  # @return [Undef]
```

Comment:
I *think* this is what I'm supposed to be doing. https://puppet.com/docs/puppet/5.5/puppet_strings_style.html#functions
> If the function has more than one potential return type, specify a @return tag for each. Begin each tag string with ‘if’ to differentiate between cases.

But the output only uses the documentation for `String`.

Reply:
You're right, it's not implemented. I'd remove the duplicate `@return` and merge the two so we at least have matching `REFERENCE.md` with the code.

Reply:
Not sure if it was ever implemented and this is a regression?  Seems odd to put something in the official style-guide if it doesn't work.

Reply:
FWIW, if I generate html docs, it doesn't use the `@return` tag at all.

Reply:
Things like javadoc also only have a single `@return` so as a develop I'm surprised about this. Then again, the java type system doesn't allow `Variant`.